internals/plan/plan.go

```diff
@@ -32,6 +32,31 @@ import (
 	"github.com/canonical/pebble/internals/osutil"
 )
 
+// LayerSectionExtension allows the plan layer schema to be extended without
+// adding centralised schema knowledge to the plan library.
+type LayerSectionExtension interface {
```

Comment:
Question from Ben: Could this simply become ```plan.Extension``` ?

Reply:
`SectionExtension` seems good. This is hooked into the system for a particular field, which we call a section, so it makes sense.  The function name should be renamed too, as RegisterSectionExtension, so we have the same terminology throughout the surface.

Reply:
Applied ```SectionExtension``` name. Resolving.